pydatastructs/trees/binary_trees.py

```diff
@@ -1053,3 +1054,86 @@ def get_sum(self, left_index, right_index):
+
+        """
+        def _zig(self,x,p):
+                if self.tree[p].left == x:
+                        super(SplayTree, self)._right_rotate(p, x)
+                else:
+                        super(SplayTree, self)._left_rotate(p, x)
+
+        def _zig_zig(self,x,p):
+                super(SplayTree, self)._right_rotate(self.tree[p].parent, p)
+                super(SplayTree, self)._right_rotate(p, x)
+
+        def _zig_zag(self,x,p):
+                super(SplayTree, self)._left_rotate(p, x)
+                super(SplayTree, self)._right_rotate(self.tree[x].parent, x)
+
+        def _zag_zag(self,x,p):
+                super(SplayTree, self)._left_rotate(self.tree[p].parent, p)
+                super(SplayTree, self)._left_rotate(p, x)
+
+        def _zag_zig(self,x,p):
+                super(SplayTree, self)._right_rotate(p, x)
+                super(SplayTree, self)._left_rotate(self.tree[x].parent, x)
+
+        def splay(self,x,p):
+                while self.tree[x].parent is not None:
+                        if self.tree[p].parent is None:
+                                self._zig(x, p)
+                        elif self.tree[p].left == x and self.tree[self.tree[p].parent].left == p:
+                                self._zig_zig(x, p)
+                        elif self.tree[p].right == x and self.tree[self.tree[p].parent].right == p:
+                                self._zag_zag(x, p)
+                        elif self.tree[p].left == x and self.tree[self.tree[p].parent].right == p:
+                                self._zag_zig(x, p)
+                        else:
+                                self._zig_zag(x, p)
+                        p = self.tree[x].parent
+
+        def insert(self,key,x):
+                super(SelfBalancingBinaryTree, self).insert(key, x)
+                e, p = super(SelfBalancingBinaryTree, self).search(x, parent=True)
+                self.tree[self.size-1].parent = p;
+                self.splay(e, p)
+
+        def delete(self,x):
+                b = super(SelfBalancingBinaryTree, self).delete(x, balancing_info=True)
+                self.splay(b, self.tree[b].parent)
+                return True
+
+        def join(self, other):
+                maxm = self.root_idx
+                while self.tree[maxm].right is not None:
+                    maxm = self.tree[maxm].right
+                self.splay(maxm, self.tree[maxm].parent)
+                other.split_array(self, other.tree[other.root_idx].data)
+
+        def split(self, x):
+                e, p = super(SelfBalancingBinaryTree, self).search(x, parent=True)
+                self.splay(e, p)
+                other = SplayTree(None, None)
+                if self.tree[self.root_idx].right is not None:
+                    self.split_array(other, self.tree[self.tree[self.root_idx].right].data)
+                    self.tree[self.root_idx].right = None
+                return other
+
+        def split_array(self, sec, x):
```

Comment:
Please add docstring for this method as this isn't common among various Tree classes.

Reply:
Done.